### DISCUSSION TITLE
what's the alternative API for `onStartShouldSetResponderCapture` and `onMoveShouldSetResponderCapture`

### DISCUSSION THREAD

**Discussion by chj-damon** [software-mansion/react-native-gesture-handler#1784] [Q&A]
I have a scrollview and inside the scrollview are TouchableOpacities.
I want to disable these TouchableOpacities when scrolling.

```
const scrolling = useSharedValue(false);

const gestureHandler = useAnimatedGestureHandler({
  onStart(evt, ctx) {
    // console.log('start');
    scrolling.value = true;
    return false;
  },
  onActive(evt) {
    // console.log(evt.translationY, 'active');
  },
  onEnd(evt) {
    scrolling.value = false;
    // console.log(evt.translationY, 'end');
  },
});

<PanGestureHandler
      ref={panRef}
      simultaneousHandlers={scrollRef}
      shouldCancelWhenOutside={false}
      enableTrackpadTwoFingerGesture
      onGestureEvent={gestureHandler}
    >
      <Animated.View style={{ flex: 1 }} onStartShouldSetResponderCapture={() => scrolling.value}>
        <NativeViewGestureHandler ref={scrollRef} simultaneousHandlers={panRef}>
          <Animated.ScrollView bounces scrollEventThrottle={1}>
            {children}
          </Animated.ScrollView>
        </NativeViewGestureHandler>
      </Animated.View>
    </PanGestureHandler>
```

`onStartShouldSetResponderCapture` will always return true, which I can't activiate these TouchableOpacities any more.

I want to know if there's a way in `react-native-gesture-handler` that I can do that?

**Comment by JasonCareter** [comment]
最新的API中、continues类型的Gesture有 manualActivation 回调、里面的state 可以用来手动fail掉Gesture。像RN的这种自定义前置条件来决定手势的响应、似乎是没得的。RNGH提供了 waitFor 和 simultaneous 来决定手势的响应顺序或同时响应。像你上面提到的这种应该是 点击事件 需要 waitFor 滚动手势响应结束才执行，ScrollView的所有子组件都要添加GestureDetector, waitFor滚动手势。

**Reply by chj-damon** [reply]
那岂不是特别麻烦？我只是想要一个下拉刷新的效果

**Reply by JasonCareter** [reply]
你是想做自定义下拉刷新吗？自定义下拉刷新在PanGesture中 START 事件中监听列表滚动偏移值, 为0时将pan的 translationY 更新到刷新指示组件的 translationY 上就行了, 另外用 interpolate 来限制其偏移范围。

**Reply by chj-damon** [reply]
是的，我是想实现自定义下拉刷新的效果。类似下面的结构：
```code
<PullToRefresh>
    <ScrollView />
   或者
   <FlatList />
  或者
  <SectionList />
</PullToRefresh>
```

一方面要保留滚动列表自带的能力，另一方面又可以在任何地方实现下拉加载（上拉刷新），具体实现上还挺麻烦的，滚动列表自带的滚动效果跟GestureHandler各种冲突。。。
